dll/win32/dbghelp/symbol.c

```diff
@@ -19,7 +19,9 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
  */
 
+#ifndef NONAMELESSUNION /* Required to avoid a warning because ROS CMake sometimes defines it on the command line */
```

Comment:
I see that the problem happens only when compiling dbghelp as a host-tool library: from its cmakefile:
```
if(NOT CMAKE_CROSSCOMPILING)
    add_definitions(-DDBGHELP_STATIC_LIB -DNONAMELESSUNION)
```